### ISSUE TITLE
Provide users an "Contact us" button

### ISSUE THREAD

**Issue by bradtaylor** [DataBiosphere/saturn-ui#735]
Users need a way to contact our support team from within the app.

The support team is using Zendesk to respond to user requests. They have created a dedicated support email for Saturn users. We also want to provide a button within the app from which users can email help.

We need to design the affordances for emailing help and integrate Zendesk's prebuilt solution. From what Geraldine has told me, this should be able to automatically capture information like console errors. See Brad to take a look at the support team's Zendesk rollout plan.

There is an affordance in current mocks for "report a bug". I think this should be a more general "Contact us", but I'm happy to listen to arguments here.
https://projects.invisionapp.com/d/main#/console/13746874/286545629/preview

**Comment by bradtaylor** [comment]
Check in with Comms to see if this is blocked on them

**Comment by katevoss** [comment]
This is no longer blocked by #740

**Comment by bradtaylor** [comment]
Default fields are name, email, "how can we help you", and attachments. You can create custom fields. 

Kate added a subject field (bug, feature request, etc). I think we'll want to keep that, since the comms flow that Ilyana showed me has some branches that hinge on these categories.

I think we can suppress name and email since we have those from the logged in user. We can surface email if people tell us they want a different one (we do make it possible on registration for users to provide a "contact email" instead of the user id. THAT'S the one we should be using).

I think we should suppress attachments. I imagine we'll want to scrape things like any console errors.

**Comment by bradtaylor** [comment]
Based on the general support flowchart, I'm suggesting we capture the request type (bug, feature request, general question, documentation request), "how can we help you", and scrape any console errors.

We should plug the email in automatically as the user's sam contact email (NOT the login ID). If that's not possible, we should include an "email" field defaulted with login ID.

**Comment by katevoss** [comment]
- [ ] add Contact Us to left main navigation
- [ ] use ? help icon
- [ ] form is blue from the Workspace nav
- [ ] text on form and buttons is white
- [ ] add subject field: Question, Feature Request, Documentation Request, Bug (default)
- [ ] form pops up in righthand corner
- [ ] Scrape console errors
- [ ] include user's current page
- [ ] auto-fill user's contact email, if none then login email
- [ ] invisibly include user's name

**Comment by katevoss** [comment]
Browser support: @bradtaylor what browsers do we want to make sure Terra works well on? How many versions?

Within Zendesk ticket forms when a field in the form is required, it does not work in Safari. I haven't investigated more but it sounds like a user in Safari can submit a form without all of the required fields. I'm assuming Safari is a required browser.

**Comment by katevoss** [comment]
Fields:
- [ ] _hidden_: name
- [ ] email: default contact, login, have option to edit
- [ ] description
- [ ] type: default Bug, Feature Request, Question
- [ ] _hidden_: console errors
- [ ] _hidden_: page submitted from ("Submitted from: http://localhost:3000/#browse-data")
- [ ] _hidden_: priority (dropdown)
- [ ] _hidden_: theme (text field)
- [ ] _hidden_: user status (dropdown)
- [ ] _hidden_: multiple users affected (checkbox)

**Comment by knoblett** [comment]
The form is set up on the Zendesk side. I am discussing with our Zendesk rep about getting @katevoss set up as a developer in our system, and will let you know when I have more info. In the meantime, let me know if there's anything I can help with.

**Comment by bradtaylor** [comment]
@katevoss Chrome is required. I'm not sure our level of browser support for Safari, IE. @panentheos @zarsky-broad can you comment?

We need this very soon, so let's roll it out and worry about browser support as followup

**Comment by panentheos** [comment]
Currently Chrome or FF (though I don't know how much FF testing we actually do). Safari and IE are generally lagging and/or broken and should be considered nice-to-have for now.

**Comment by katevoss** [comment]
@knoblett something to consider, should we have a dev instance of Zendesk set up so that when our team is testing things out that we won't be sending emails and tickets to the official prod version of Zendesk?

Also, I hit a rate limit of the number of requests to create a ticket. Do you know if there is a limit in production for how many times we can create requests? I can look into it but I wasn't sure if you'd heard anything from sales at Zendesk.

**Comment by katevoss** [comment]
Some info from Zendesk on our API rate limit: https://develop.zendesk.com/hc/en-us/articles/360001068967--API-UPDATE-Account-wide-API-rate-limit-rollout

**Comment by knoblett** [comment]
Yes; That would be a good idea to use a dev instance of Zendesk for testing. We can do it for free for as long as your trial will last, but long-term it would be better from a financial aspect to work in the main Zendesk we have. If the latter is the case, let me know and I will communicate it to our support agents so they'll be able to distinguish between real requests and test requests. 

Does the link you put in about APIs answer your question? If not, I can introduce you to Hunter, our Zendesk rep, to get that question answered.

**Comment by katevoss** [comment]
@bradtaylor We have an opportunity for a cute, whimsical, pleasant message once a user submits a support request.

Something like: 
>Successfully sent. Our elves will get back to you shortly.

or 
>All done! Check your email to update your support request.

>Thanks for contacting us. We'll get back to you soon.

>Success! We will be in touch by email.

>Success! We'll put Terry on the case!

What do you think?
